test/fuzz/fuzz_runner.cc

```diff
@@ -30,6 +31,12 @@ void Runner::setupEnvironment(int argc, char** argv, spdlog::level::level_enum d
   // spew too much when running under a fuzz engine.
   log_level_ =
       environment_log_level <= spdlog::level::debug ? environment_log_level : default_log_level;
+  // This needs to work in both the Envoy test shim and oss-fuzz build environments, so we can't
+  // allocate in main.cc. Instead, just create these non-PODs to live forever.
```

Comment:
Can you note also that there is no notification of tear-down, and thus no opportunity to clean up?